### PR TITLE
fix(providers): make AoT compile friendly

### PR DESCRIPTION
**Pull request by robwormald**
cc @davideast

### CODE REVIEW

**Comment by mhevery** [comment]
LGTM

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "test:watch": "karma start",
     "build": "rm -rf dist; tsc",
     "build:watch": "rm -rf dist && tsc -w",
-    "build_npm": "rm -rf dist && tsc -p tsconfig.publish.es5.json && tsc -p tsconfig.publish.es6.json && tsc -p tsconfig.publish.es6-deprecated.json",
+    "build_npm": "rm -rf dist && ngc -p tsconfig.publish.es5.json && ngc -p tsconfig.publish.es6.json",
```
Comment (davideast) [inline package.json]: 😍

File: package.json [hunk 2]
```diff
@@ -66,7 +68,7 @@
     "traceur": "0.0.96",
     "tsd": "^0.6.5",
     "typedoc": "github:jeffbcross/typedoc",
-    "typescript": "^1.8.10",
+    "typescript": "1.9.0-dev.20160627-1.0",
```
Comment (davideast) [inline package.json]: Needs to be 2.0 plz.

File: src/angularfire2.spec.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import {
   AngularFireDatabase,
   FirebaseAppConfig
 } from './angularfire2';
-import {Subscription} from 'rxjs';
+import {Subscription} from 'rxjs/Subscription';
```
Comment (davideast) [inline src/angularfire2.spec.ts]:
```js
import { Subscription } from 'rxjs/Subscription';
```

File: docs/1-install-and-setup.md [hunk 4]
```diff
@@ -122,25 +122,28 @@ Open `/src/main.ts`, inject the Firebase providers, and specify your Firebase co
...
-    apiKey: "<your-key>",
-    authDomain: "<your-project-authdomain>",
-    databaseURL: "<your-database-URL>",
-    storageBucket: "<your-storage-bucket>"
-  })
-]);
+@NgModule({
+  imports: [
+    BrowserModule,
+    AngularFireModule.forConfig(firebaseConfig)
```
Comment (davideast) [inline docs/1-install-and-setup.md]: Lets go with

```js
AngularFireModule.initializeApp(firebaseConfig)
```

File: src/auth/auth.spec.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { Observable } from 'rxjs/Observable'
 import { Observer } from 'rxjs/Observer';
 import {
   addProviders,
-  inject
+  inject,
```
Comment (davideast) [inline src/auth/auth.spec.ts]: Dangling comma

**Pull request merged**